sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -281,6 +285,10 @@ abstract class HashExpression[E] extends Expression {
       DataTypeMismatch(
         errorSubClass = "HASH_MAP_TYPE",
         messageParameters = Map("functionName" -> toSQLId(prettyName)))
+    } else if (children.exists(child => hasVariantType(child.dataType))) {
+      DataTypeMismatch(
+        errorSubClass = "VARIANT_TYPE",
```

Comment:
shall we follow `HASH_MAP_TYPE` and name it `HASH_VARIANT_TYPE`?

Reply:
Oh, that's correct. I probably got confused by another error subclass called `MAP_TYPE`. Fixed now. Resolving.